portfolio/src/test/java/com/google/sps/BookDaoTest.java

```diff
@@ -0,0 +1,214 @@
+    ArrayList<String> authors = new ArrayList<String>();
+    authors.add("Sarah J. Maas");
+    String isbn = "9781619634497";
+    int pageCount = 432;
+    ArrayList<String> categories = new ArrayList<String>();
+    categories.add("Fiction");
+    Set<String> genre = new HashSet<String>();
+    genre.add("Fantasy");
+    String publisher = "Bloomsbury Publishing USA";
+    String language = "en";
+    String publishedDate = "2017-05-02";
+    String maturityRating = "NOT_MATURE";
+    String thumbnail = "http://books.google.com/books/content?id=pLL-DAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api";
+    String infoLink = "https://play.google.com/store/books/details?id=pLL-DAAAQBAJ&source=gbs_api";
+    String description = "Looming war threatens all Feyre holds dear in the third volume of the #1 "
+        +"New York Times bestselling A Court of Thorns and Roses series. Feyre has returned to the "
+        +"Spring Court, determined to gather information on Tamlin\'s maneuverings and the invading "
+        +"king threatening to bring Prythian to its knees. But to do so she must play a deadly game "
+        +"of deceit-and one slip may spell doom not only for Feyre, but for her world as well. As war "
+        +"bears down upon them all, Feyre must decide who to trust amongst the dazzling and lethal "
+        +"High Lords-and hunt for allies in unexpected places. In this thrilling third book in the "
+        +"#1 New York Times bestselling series from Sarah J. Maas, the earth will be painted red as "
+        +"mighty armies grapple for power over the one thing that could destroy them all.";
+
+    Book.Builder builder = Book.builder().title(title).genre(genre)
+        .categories(categories).authors(authors).language(language)
+        .description(description).infoLink(infoLink).pageCount(pageCount)
+        .publishedDate(publishedDate).publisher(publisher)
+        .maturityRating(maturityRating).thumbnail(thumbnail).isbn(isbn);
+    Book expected = builder.build();
+    return expected;
+  }
+
+  private void compareBooks(Book expected, Book actual){
+    Assert.assertEquals(expected.title(),actual.title());
+    Assert.assertEquals(expected.genre(),actual.genre());
+    Assert.assertEquals(expected.categories(),actual.categories());
+    Assert.assertEquals(expected.authors(),actual.authors());
+    Assert.assertEquals(expected.language(),actual.language());
+    Assert.assertEquals(expected.description(),actual.description());
+    Assert.assertEquals(expected.infoLink(),actual.infoLink());
+    Assert.assertEquals(expected.pageCount(),actual.pageCount());
+    Assert.assertEquals(expected.publishedDate(),actual.publishedDate());
+    Assert.assertEquals(expected.publisher(),actual.publisher());
+    Assert.assertEquals(expected.maturityRating(),actual.maturityRating());
+    Assert.assertEquals(expected.thumbnail(),actual.thumbnail());
+    Assert.assertEquals(expected.isbn(),actual.isbn());
```

Comment:
This might work, instead of checking all fields:  `Assert.assertEquals(expected,actual);`. The Book class is implemented with `Autovalue` and it generates the equals method including each field.
If so, you could just remove this method.